core/src/main/java/io/grpc/internal/MessageDeframer.java

```diff
@@ -338,10 +342,10 @@ private void processHeader() {
 
     // Update the required length to include the length of the frame.
     requiredLength = nextFrame.readInt();
-    if (requiredLength < 0 || requiredLength > maxMessageSize) {
+    if (requiredLength < 0 || requiredLength > maxInboundMessageSize) {
       throw Status.INTERNAL.withDescription(String.format("Frame size %d exceeds maximum: %d. "
               + "If this is normal, increase the maxMessageSize in the channel/server builder",
```

Comment:
This can be misleading because CallOptions and service config will be able to override the message size set in the builder. Maybe just delete the second sentence?